pkg/tsdb/prometheus/custom_query_params_middleware.go

```diff
@@ -25,21 +24,18 @@ func customQueryParametersMiddleware() sdkhttpclient.Middleware {
 			return next
 		}
 
+		values, err := url.ParseQuery(customQueryParams)
+		if err != nil {
+			log.Error("Failed to parse custom query parameters, skipping middleware", "error", err)
+			return next
+		}
+
 		return sdkhttpclient.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
-			params := url.Values{}
-			for _, param := range strings.Split(customQueryParams, "&") {
-				parts := strings.Split(param, "=")
-				if len(parts) == 1 {
-					// This is probably a mistake on the users part in defining the params but we don't want to crash.
-					params.Add(parts[0], "")
-				} else {
-					params.Add(parts[0], parts[1])
+			if len(values) > 0 {
+				if len(req.URL.RawQuery) > 0 {
+					req.URL.RawQuery += "&"
 				}
-			}
-			if req.URL.RawQuery != "" {
-				req.URL.RawQuery = fmt.Sprintf("%s&%s", req.URL.RawQuery, params.Encode())
-			} else {
-				req.URL.RawQuery = params.Encode()
+				req.URL.RawQuery += values.Encode()
```

Comment:
Depending on whether we want to override and/or validate existing query values - it might be worth considering using req.URL.Query():

```
q := req.URL.Query()
for k := range values {
	q.Add(k, values.Get(k)) // or q.Set(), if we want new values to overwrite the existing ones
}
req.URL.RawQuery = q.Encode()
```